doc/dev/installation.rst

```diff
@@ -414,3 +414,82 @@ To generate html files for the documentation for Catalyst:
   make docs
 
 The generated files are located in ``doc/_build/html``
+
+Install a Frontend-Only Development Environment from TestPyPI Wheels
+--------------------------------------------------------------------
+
+It is possible to work on the source code repository and test the changes without 
+having to compile Catalyst. This is ideal for situations where the changes do not target the 
+runtime or the MLIR infrastructure, and only concern to the frontend. It basically 
+makes use of the shared libraries already shipped with the TestPyPI Catalyst wheels.
+
+Essential Steps
+^^^^^^^^^^^^^^^
+
+To activate the development environment, open a terminal and issue the following commands:
+
+.. code-block:: console
+
+  # Clone the Catalyst repository  
+  git clone --recurse-submodules --shallow-submodules https://github.com/PennyLaneAI/catalyst.git
```

Comment:
This step takes quite a while (because of how big llvm is), which led we to wonder whether we even need the submodules if this is meant to be a frontend environment only. Although having them might still be a good idea, not sure 🤔

Reply:
I have changed the "Essential Steps" sections suggesting a simple cloning. In "Special Considerations" I specify the command for a full cloning with submodules. Done.